lib/library.rb

```diff
@@ -0,0 +1,76 @@
+require 'yaml'
+require 'date'
+require 'pry-byebug'
+
+class Library
+  attr_reader :collection, :return_date
+
+  def initialize
+    @collection = load_book_collection
+  end
+
+  def book_check_out(item, patron)
+    book = find_book(item)
+    case
+    when book == 'no book'
+      'Library does not have this book'
+    when book[:available] == false
+      'Book is not available. It will be returned ' +
+        (@collection[book[:index]][:return_date]).to_s
+    else
+      perform_transaction(book, patron)
+    end
+  end
+
+  def available_books
+    avail_books = []
+    @collection.each do |book|
+      avail_books << book if book[:available] == true
+    end
+    avail_books
+  end
+
+
+  private
+
+  def load_book_collection
+    YAML.load_file('./lib/data.yml')
+  end
+
+  def update_book_collection
+    File.open('./lib/data.yml', 'w') { |f| f.write collection.to_yaml }
+  end
+
+  def perform_transaction(book, patron)
```

Comment:
I'd think of a better name for this method.. `perform_transaction` reminds me of a bank transaction (re: atm)